src/query.js

```diff
@@ -68,22 +95,34 @@ export function createHashString(mapSettings) {
 
   let nonMapSettings = Object.fromEntries(
     Object.entries(newMapSettings)
-      .filter(([k, v]) => !mapLocationKeys.includes(k))
+      .filter(([k, v]) => !mapLocationKeys.includes(k) && k !== 'locations')
       .map(([k, v]) => [k, jsonKeys.includes(k) ? JSON.stringify(v) : v])
   );
 
   nonMapSettings = cleanSettings(nonMapSettings);
 
-  return toQueryString({
-    map: [
-      round(newMapSettings.zoom, 2),
-      round(newMapSettings.center.lat, 4),
-      round(newMapSettings.center.lng, 4),
-      round(newMapSettings.pitch, 1),
-      round(newMapSettings.bearing, 1),
-    ].join('/'),
+  let updatedSettings = {
     ...nonMapSettings,
-  });
+  };
+
+  if (linkedLocations) {
+    updatedSettings = {
+      map: encodeMapParams(newMapSettings),
+      ...updatedSettings,
+    };
+  } else {
+    updatedSettings = {
+      locations: JSON.stringify(
+        newMapSettings.locations.map(item => ({
+          ...item,
+          location: encodeMapParams(item.location),
+        }))
+      ),
+      ...updatedSettings,
+    };
+  }
```

Comment:
The url will always show _either_ `map` _or_ `locations` for the location params.